### DISCUSSION TITLE
Discount Coupon: no error if Purchase Total is set

### DISCUSSION THREAD

**Discussion by stevenvandemoortele** [craftcms/commerce#2334] [Ideas]
### Description
Commerce doesn't return an error when trying to apply a discount code to a cart that has a lower purchase total than the purchase total defined in the coupon

### Steps to reproduce

1. Create coupon with a Purchase Total value
2. Add purchasable to cart with lower Purchase Total than defined in the coupon
3. Try to add coupon code
4. cart.getFirstError('couponCode') doesn't have an error
5. craft.app.session.getFlash('error') is null
6. craft.app.session.getFlash('notice') has 'Cart updated.' as value

### Additional info

- Craft version: 3.1.27
- Plugins & versions: Commerce 2.1.6.1

**Comment by nfourtythree** [comment]
Hey @stevenvandemoortele 

Due to the structure of discounts, this isn't technically an error as eventually if you keep adding items it will match. It also has implications if you are doing multiple things in one request e.g. updating a line item quantity and adding a coupon code. We wouldn't want the cart failing and confusing people.

We are looking a refactor/update to the discount functionality in the next major release of Commerce, at which point there will be better support for this type of situation.

You can get information on the state of these types of coupons by writing some custom twig code or custom code in a plugin/module. For that, it would be a good idea to look at the following code: https://github.com/craftcms/commerce/blob/develop/src/adjusters/Discount.php#L143-L184

Thanks!
